### DISCUSSION TITLE
Stacking notifications of the same type

### DISCUSSION THREAD

**Discussion by s-wel** [Avaiga/taipy#173] [Q&A] [answered]
Hi, I want to show several notifications of the same type (success) shortly created after another in a for-loop. I can only see a maximum of two notifications (and not four as I expected). Is there a limit to how many notifications can be visualized at a time?

**Comment by FlorianJacta** [comment]
Hi,

There should be a maximum of 5 notifications:

```python
from taipy.gui import Gui, notify

def update(state):
    for i in range(10):
        notify(state, 'scenario', "Success")

Gui('<|Push|button|on_action=update|>').run()
```
Do you have a code to replicate it?

**Reply by s-wel** [reply]
I have a snippet (see below) that I use to retrieve the logs from Docker containers via the Python Docker SDK. 
I execute it when the user clicks on a button. If I run it, I can only see either 2 success or 2 info notifications. I have four names in the "names" list and should get four notifications in total. If I click the button several times in a row I get more notifications that also stack properly but it seems some notifications are simply not sent.

```
for name in names:

        container = client.containers.get(name)

        if container:

            dfs[name] = read_logs_between_timestamps(
                container,
                filters.get('since', datetime.now()),
                filters.get('until', datetime.now())
            )

            if dfs[name].empty:
                notify(state, "info", f"{name} has no data for this period.")
            else:
                notify(state, "success", f"{name} data retrieved.")

        else:
            notify(state, "error", f"No container {name} found")
            dfs[name] = DataFrame()
```

**Reply by s-wel** [reply]
I also realized that the behavior differs in Firefox and Chrome. In Chrome it works a bit better but still some notifications never appear.

**Reply by FlorianJacta** [reply]
Interesting! I am trying to replicate the behavior. Do you have this behavior with the code below?

```python
from taipy.gui import Gui, notify

integers = [6, 0, 7, 2, 0, 6, 9, 2, 0, 0]

def update(state):
    for integer in integers:
        if integer > 0:
            if integer < 5:
                notify(state, "info", f"{integer} has no data for this period.")
            else:
                notify(state, "success", f"{integer} data retrieved.")

        else:
            notify(state, "error", f"No container {integer} found")


Gui('<|Update|button|on_action=update|>').run()
```

**Reply by s-wel** [reply]
I tested your code and it seems that sometimes I see 3 notifications and sometimes 4 or 5. Both in Firefox and Chrome. Tested with a notebook deployment (no containers or server).

**Reply by FlorianJacta** [reply]
We will investigate it. The issue is hard to replicate:. https://github.com/Avaiga/taipy-gui/issues/777

**Reply by FlorianJacta** [reply, marked as answer]
The issue is closed, and the PR is merged into the 'develop' branch. You will be able to test it in the next release.
